src/pages/journal/components/test-outcome/test-outcome.ts

```diff
@@ -74,23 +72,74 @@ export class TestOutcomeComponent {
     return this.testStatus === TestStatus.Started;
   }
 
-  showRekeyModal = (): void => {
+  displayRekeyModal = (): void => {
     const options = { cssClass: 'mes-modal-alert text-zoom-regular' };
     this.modal = this.modalController.create('JournalRekeyModal', {}, options);
+    this.modal.onDidDismiss(this.onModalDismiss);
     this.modal.present();
   }
 
+  startTest(rekey: boolean) {
+    this.store$.dispatch(new StartTest(this.slotDetail.slotId));
+    if (rekey) {
+      this.store$.dispatch(new MarkAsRekey(this.slotDetail.slotId));
+    }
+    this.navController.push(COMMUNICATION_PAGE);
+  }
+
   onModalDismiss = (event: ModalEvent): void => {
     switch (event) {
       case ModalEvent.CANCEL:
         this.modal.dismiss();
         break;
       case ModalEvent.START:
-        // todo: Logic to be impliment later
+        this.startTest(false);
         break;
       case ModalEvent.REKEY:
-        // todo: Logic to be impliment later
+        this.startTest(true);
         break;
     }
   }
+
+  rekeyTest() {
+    if (this.testStatus === TestStatus.Booked) {
+      this.startTest(true);
```

Comment:
How come the test is a rekey if it was booked? Shouldn't this use the logic from the `showRekeyModal()` method?

Reply:
The `showRekeyModal()` logic is only for tests today that are before the `start time + duration of test`. E.g. if it's a 10am test and 60 min duration the modal should show from 11am onwards. The blue rekey button is displayed if the test is on a previous day.